mq/handlers.go

```diff
@@ -145,13 +145,29 @@ func UpdateHost(client mqtt.Client, msg mqtt.Message) {
 		var sendPeerUpdate bool
 		switch hostUpdate.Action {
 		case models.UpdateHost:
+			if servercfg.GetBrokerType() == servercfg.EmqxBrokerType {
+				// create EMQX credentials for host if it doesn't exists
+				if _, err := logic.GetHost(currentHost.ID.String()); err != nil {
+					if err := CreateEmqxUser(currentHost.ID.String(), currentHost.HostPass, false); err != nil {
+						logger.Log(0, "failed to add host credentials to EMQX: ", currentHost.ID.String(), err.Error())
```

Comment:
how will client even publish msg if there is no user created already for the host..
this check here for host doesn't make sense to me

Reply:
done